deploy/deployUtils.ts

```diff
@@ -486,13 +494,58 @@ export async function deploySwapFlashLoanPools(
       abi: (await get("LPToken")).abi, // LPToken ABI
       address: lpTokenAddress,
     })
-    // verify contract
+    // register new pools
+    if (newDeploypools.length > 0) {
+      await registerPools(hre, newDeploypools)
+      // If new pools require a gauge deploy
+      if (pool.deployGauge) {
+        deployLiquidityGauge(hre, lpTokenName, lpTokenAddress)
+      }
+    }
+    // verify contracts
     await verifyContract(hre, poolName)
+    await verifyContract(hre, lpTokenName)
+  }
+}
+
+/**
+ * @notice deploys a LiquidityV5Gauges for each given swap LPToken element that has not yet been deployed.
+ * Deploys and saves deployment for the associated gauge.
+ * @param lpToken string of deployed LPToken name
+ * @param lpTokenAddress string of deployed LPToken address
+ */
+export async function deployLiquidityGauge(
+  hre: HardhatRuntimeEnvironment,
+  lpToken: string,
+  lpTokenAddress: string,
+) {
+  const { deployments, getNamedAccounts } = hre
+  const { deploy, get, getOrNull } = deployments
+  const { deployer } = await getNamedAccounts()
+  const gaugeName = `LiquidityGaugeV5_${lpToken}`
+  const gaugeContract = await getOrNull(gaugeName)
+  if (gaugeContract) {
+    console.log(
+      `Gauge ${gaugeName} already deployed at ${gaugeContract.address}`,
+    )
   }
-  // register new pools
-  if (newDeploypools.length > 0) {
-    await registerPools(hre, newDeploypools)
+  if (!gaugeContract) {
```

Comment:
This is unnecessary if check since we already know gaugeContract is either null or undefined from L527 check.